doc/src/manual/strings.md

```diff
@@ -681,7 +681,7 @@ of the substring that matches, but perhaps we want to capture any non-blank text
 character. We could do the following:
 
 ```jldoctest
-julia> m = match(r"^\s*(?:#\s*(.*?)\s*$|$)", "# a comment ")
+julia> m = match(r"^\s*(?:#\s*(.*?)\s*$)", "# a comment ")
```

Comment:
this changes the meaning of regex ("\s*" matches the former and not the latter), although I don't know if the difference is material (the regex is only a rough approximation of the operation it describes, and would be tripped up by strings, character literals, or multi-line comments)

Reply:
The `*` after the `\s` is for zero or more, so the `$|$` alternation will never be of use.